### PR TITLE
feat: drop support of php 7

### PR DESCRIPTION
**Pull request by Chris53897**
i am not sure how to handle the failing `--prefer-lowest` run.
Is symfony 3.4 still the desired minimum version?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## 3.2.0 (2022-xx-xx)
 
+* Increased minimum PHP version to 8.0 (to be consistent with KnpMenu 3.5)
```
Comment (stof) [inline CHANGELOG.md:3]: We should not have 2 `Increased minimum PHP version to` in the changelog for the same version of the bundle.

**Review by stof — COMMENTED** [review]
Rather than spending time fixing compat issues for Symfony 3.4 (which indeed never received the full compat fixes for PHP 8 as it reached end of bug fixes at the time of the PHP 8.0 release), I suggest dropping support for Symfony 3.4 (and probably also 4.4 at the same time as it also reached end of bug fix support)

File: CHANGELOG.md (hunk 1 above)
Comment (garak) [reply CHANGELOG.md:3]: +1 for dropping Symfony 4.4

Comment (garak) [reply CHANGELOG.md:3]: 3.2.0 has already been released, we forgot to update the changelog.
Please fix it, and the new entry as 3.3.0

**Review by garak — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Chris53897) [reply CHANGELOG.md]: I changed it with the help of https://github.com/KnpLabs/KnpMenuBundle/releases . But as there is no KnpMenu release 3.4 for php 7.4 (as planned)
https://github.com/KnpLabs/KnpMenu/releases

Please have a look at the new version.

**Review by stof — CHANGES_REQUESTED** [review]
The logic about bundle inheritance in BuilderAliasProvider can also be removed, as this feature was dropped in Symfony 4.

**Comment by Chris53897** [comment]
I changed it. But i am not sure if there are more changes required

**Review by derrabus — APPROVED** [review]

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Provider/BuilderAliasProvider.php [hunk 2]
```diff
@@ -94,11 +84,7 @@ private function getBuilder(string $bundleName, string $className): object
             $bundles = [];
 
             $allBundles = $this->kernel->getBundle($bundleName, false);
```
[the remark this answers is not in the document]
Comment (Chris53897) [reply src/Provider/BuilderAliasProvider.php]: ok, done

File: src/Provider/BuilderAliasProvider.php [hunk 3]
```diff
@@ -94,11 +84,7 @@ private function getBuilder(string $bundleName, string $className): object
...
-
-            // In Symfony 4, bundle inheritance is gone, so there is no way to get an array anymore.
-            if (!\is_array($allBundles)) {
-                $allBundles = [$allBundles];
-            }
+            $allBundles = [$allBundles];
```
[the remark this answers is not in the document]
Comment (Chris53897) [reply src/Provider/BuilderAliasProvider.php]: Thanks for the help. I hope it did the changes correct.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
-## 3.2.0 (2022-xx-xx)
+## 3.3.0 (2023-05-xx)
 
-* Increased minimum PHP version to 7.4 (to be consistent with KnpMenu 3.4)
+* Increased minimum PHP version to 8.0 (to be consistent with KnpMenu 3.4)
+* Dropped support for Symfony 3 and 4
+
+## 3.2.0 (2021-10-30)
+
+* Add support for Symfony 6
```
Comment (garak) [inline CHANGELOG.md:8]: This should be "Added", not "Add".

**Review by garak — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (Chris53897) [reply CHANGELOG.md]: Done

**Review by garak — APPROVED** [review]

**Pull request merged**